identify-ref-leaks/wrap-for-leakscript.sh

```diff
@@ -0,0 +1,233 @@
+#  This is a wrapper script which will run the stap command for detecting
+#  leaks. It can be run in interactive as well as non-interactive mode.In
+#  interactive mode it will ask few options after running this script. In
+#  non-interactive mode, user has to provide the arguments while running this
+#  script.
+
+#!/bin/bash
+
+status=1
+func_name=""
+func_name_unref=""
+pid_gluster=""
+output_file=""
+probe_time=900
+module=""
+str_length=900
+file_size=40
+max_no_of_files=100
+
+check_fun_name()
+{
+        #matching ref function with its counter unref function
+
+        if [[ $status == 0 ]]
+        then
+                case $func_name in
+                        "dict_ref");;
+                        "inode_ref");;
+                        "fd_ref");;
+                        *)
+                        echo "Function name can be dict_ref/inode_ref/fd_ref"
+                        exit;;
+                 esac
+        fi
+
+        if [[ $func_no == 1 || $func_name == "dict_ref" ]]
+        then
+                func_name="dict_ref"
+                func_name_unref="dict_unref"
+        elif [[ $func_no == 2 || $func_name == "inode_ref"  ]]
+        then
+                func_name="__inode_ref"
+                func_name_unref="__inode_unref"
+        elif [[ $func_no == 3 || $func_name == "fd_ref" ]]
+        then
+                func_name="__fd_ref"
+                func_name_unref="__fd_unref"
+        else
+                echo  "\nWrong choice. Function name can be
+                dict_ref/inode_ref/fd_ref"
+                exit
+        fi
+}
+
+fun_name()
+{
+	echo -e "\nEnter the no. for the function which you wants to probe."
+	echo -e  "Press\n1. dict_ref\n2. inode_ref\n3. fd_ref"
+	read func_no
+
+        check_fun_name
+}
+
+process_id()
+{
+    echo -e "\nEnter Process Id :"
+    read pid_gluster
+}
+
+input_output_file()
+{
+        #creating output file, name based on epoch time
+
+        dir=/var/run/gluster/leak-output
+        if [ ! -d $dir ]
+        then
+               mkdir -p $dir
+        fi
+
+        current_time=`date +%s`
+	output_file="trace-"$current_time
+	echo -e "\noutput file : $output_file at "$dir
+
+        #input file
+        input_file="./ref-leak-identify.stp"
+
+}
+
+check_probe_time()
+{
+        temp_time=$1
+
+        if [[ -z $temp_time ]]
+        then
+                probe_time=900
+        else
+                probe_time=$(( temp_time*60 ))
+        fi
+}
+
+time_int()
+{
+	echo -e "\nEnter probing time interval in minutes : "
+	echo -e "( Default is 15 minutes )"
+	read probe_time
+        check_probe_time probe_time
+}
+
+
+
+select_module()
+{
+        #Checking installation is from source or by rpm and assigning
+        #xlators accordingly.
+
+        gluster_version=`gluster --version | cut -d ' ' -f 2 | grep dev`
+
+        rpm_res1=`rpm -q glusterfs`
```

Comment:
`pkg-config` is only available for developer packages that include headers (well, `.pc` files).

Doing this for `rpm` is fine, but it would be better to add an error message in case `rpm` is not available.